blocklib/picoscope/Picoscope.hpp

```diff
@@ -736,7 +738,8 @@ struct Picoscope : public PicoscopeBlockingHelper<TPSImpl, gr::DataSetLike<T>>::
     void processDriverDataStreaming(std::size_t nSamples, std::size_t offset, gr::InputSpanLike auto& timingInSpan, gr::OutputSpanLike auto& digitalOutSpan, std::span<TOutSpan>& outputs, std::chrono::nanoseconds acquisitionTime)
     requires(acquisitionMode == AcquisitionMode::Streaming)
     {
-        std::size_t availableSamples = calculateAvailableOutputs(nSamples, digitalOutSpan, outputs);
+        std::size_t       availableSamples = calculateAvailableOutputs(nSamples, digitalOutSpan, outputs);
+        const std::size_t nDroppedSamples  = (nSamples >= availableSamples) ? nSamples - availableSamples : 0uz;
```

Comment:
this check is redundant with l730, which does the reporting of the error, this should imo be moved here to only have this in one place. Otherwise the condition might change slightly (it is already `<` vs `>=`) and cause confusion,